dynamo: Count number of opcodes processes

c00w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #147149

This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

c00w committed:
dynamo: Count number of opcodes processes

This gives us a decent proxy for graph size. Sending just the recording to 1)
see if there are perf issues, and 2) to trigger discussion

[ghstack-poisoned]

c00w committed:
Update on "[WIP] dynamo: Count number of opcodes processes"


This gives us a decent proxy for graph size. Sending just the recording to 1)
see if there are perf issues, and 2) to trigger discussion

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "[WIP] dynamo: Count number of opcodes processes"


This gives us a decent proxy for graph size. Sending just the recording to 1)
see if there are perf issues, and 2) to trigger discussion

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "[WIP] dynamo: Count number of opcodes processes"


This gives us a decent proxy for graph size. Sending just the recording to 1)
see if there are perf issues, and 2) to trigger discussion

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

masnesral commented:
> Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

Yeah, can we either bump the counter via metrics context or clear the counters? Aren't we viewing everything in the CompilationMetrics (dynamo_compile table) as per compileId? I think it would be easy to get confused if it's cumulative across compileIds.

jansel commented on test/dynamo/test_utils.py:
```diff
@@ -443,6 +445,33 @@ def test_dynamo_timed(self, mock_time, mock_time_ns):
+        with mock.patch("torch._dynamo.utils.log_compilation_event") as log_event:
+            torch.compile(test1)(torch.randn(10, 10))
+            compilation_events = [arg[0][0] for arg in log_event.call_args_list]
+        self.assertEqual(compilation_events[0].ir_count, 10)
```
I don't believe this will be stable across python versions, can we make sure these tests don't fail on different Python versions?

c00w replied:
Yeah this isn't stable, and in fact py3.13 is failing on the PR (I tested with 3.10). Let me make this test more resilient. (I'll make it fail on unknown python versions so we update it + hardcode existing versions).

c00w replied:
Updated to explicitly encode versions, let me know if that's too explicit, ad we shoudl just check - ir >0 and second < first.

jansel commented on torch/_dynamo/symbolic_convert.py:
```diff
@@ -1102,6 +1102,7 @@ def step(self):
             return False
         self.current_instruction = inst = self.instructions[ip]
         self.instruction_pointer = ip + 1
+        counters["dynamo_step_count"]["total"] += 1
```
Do we need to increment this every bytecode?  Why not just at the end and at jumps so the cost is not O(n).

c00w replied:
I think I updated this correctly, to do this at the end and when jumping :). (also let me know if you have good examples of code with jumps so I can help validate I did it correctly).

jansel reviewed (changes requested):
The cumulative count seems a bit weird.

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w commented:
Updated to use metrics_context (letting tests run to confirm that we're always within a metrics_context).

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
None on "dynamo: Count number of opcodes processes"


This gives us a decent proxy for how big of a graph we functionally had to parse.

Note that this is a cummulative counter. If people feel strongly, I can either write into the dynamo_timed datasets with metrics contexts, or clear the counters / write a counter per frame id as well.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w commented:
@pytorchbot merge
